app/assets/javascripts/discourse/app/mixins/composer-video-thumbnail-uppy.js

```diff
@@ -43,73 +43,81 @@ export default Mixin.create(ExtendableUploader, UppyS3Multipart, {
       .substring(video_src.lastIndexOf("/") + 1)
       .split(".")[0];
 
-    // Wait for the video element to load, otherwise the canvas will be empty
-    video.oncanplay = () => {
-      let canvas = document.createElement("canvas");
-      let ctx = canvas.getContext("2d");
-      let videoHeight, videoWidth;
-      videoHeight = video.videoHeight;
-      videoWidth = video.videoWidth;
-      canvas.width = videoWidth;
-      canvas.height = videoHeight;
-
-      ctx.drawImage(video, 0, 0, videoWidth, videoHeight);
-
-      // upload video thumbnail
-      canvas.toBlob((blob) => {
-        this._uppyInstance = new Uppy({
-          id: `screenshot-placeholder`,
-          meta: {
-            upload_type: `thumbnail`,
-            video_sha1,
-          },
-          autoProceed: true,
-        });
-
-        if (this.siteSettings.enable_upload_debug_mode) {
-          this._instrumentUploadTimings();
-        }
-
-        if (this.siteSettings.enable_direct_s3_uploads) {
-          this._useS3MultipartUploads();
-        } else {
-          this._useXHRUploads();
-        }
-        this._uppyInstance.use(DropTarget, { target: this.element });
-
-        this._uppyInstance.on("upload", () => {
-          this.set("uploading", true);
-        });
+    let canvas = document.createElement("canvas");
+    if (canvas.getContext) {
```

Comment:
Mostly whitespace changes, see the diff without them:

https://github.com/discourse/discourse/pull/20644/files?diff=unified&w=1